lte/gateway/c/oai/tasks/amf/nas_proc.cpp

```diff
@@ -303,38 +307,39 @@ static int identification_t3570_handler(
         LOG_AMF_APP, "T3570 timer   timer id %d ue id %d\n",
         ident_proc->T3570.id, ident_proc->ue_id);
     ident_proc->T3570.id = NAS5G_TIMER_INACTIVE_ID;
-  }
-  /*
-   * Increment the retransmission counter
-   */
-  ident_proc->retransmission_count += 1;
-  OAILOG_ERROR(
-      LOG_AMF_APP, "Timer: Incrementing retransmission_count to %d\n",
-      ident_proc->retransmission_count);
-
-  if (ident_proc->retransmission_count < IDENTIFICATION_COUNTER_MAX) {
     /*
-     * Send identity request message to the UE
+     * Increment the retransmission counter
      */
+    ident_proc->retransmission_count += 1;
     OAILOG_ERROR(
-        LOG_AMF_APP, "Timer: IDENTIFICATION_COUNTER_MAX =  %d\n",
-        IDENTIFICATION_COUNTER_MAX);
-    OAILOG_ERROR(
-        LOG_AMF_APP,
-        "Timer: timer has expired Sending Identification request again\n");
-    amf_identification_request(ident_proc);
-  } else {
-    /*
-     * Abort the identification procedure
-     */
-    OAILOG_ERROR(
-        LOG_AMF_APP,
-        "Timer: Maximum retires done hence Abort the identification "
-        "procedure\n");
-    return -1;
+        LOG_AMF_APP, "Timer: Incrementing retransmission_count to %d\n",
+        ident_proc->retransmission_count);
+
+    if (ident_proc->retransmission_count < IDENTIFICATION_COUNTER_MAX) {
+      /*
+       * Send identity request message to the UE
+       */
+      OAILOG_ERROR(
+          LOG_AMF_APP, "Timer: IDENTIFICATION_COUNTER_MAX =  %d\n",
+          IDENTIFICATION_COUNTER_MAX);
+      OAILOG_ERROR(
+          LOG_AMF_APP,
+          "Timer: timer has expired Sending Identification request again\n");
+      amf_identification_request(ident_proc);
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNok);
+    } else {
+      /*
+       * Abort the identification procedure
+       */
+      OAILOG_ERROR(
+          LOG_AMF_APP,
+          "Timer: Maximum retires done hence Abort the identification "
+          "procedure\n");
+      amf_remove_ue_context(ue_context);
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNerror);
+    }
   }
-#endif /*  TIMER_CHANGES_REVIEW */
-  return 0;
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNerror);
```

Comment:
it is possible to drive these retransmissions through state machine. it is important to drive it all through state machine.